server/src/test/java/org/opensearch/index/store/CompositeDirectoryTests.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakFilters;
+
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.store.remote.file.CleanerDaemonThreadLeakFilter;
+import org.opensearch.index.store.remote.file.OnDemandBlockSnapshotIndexInput;
+import org.opensearch.index.store.remote.filecache.CachedIndexInput;
+import org.opensearch.index.store.remote.filecache.FileCache;
+import org.opensearch.index.store.remote.filecache.FullFileCachedIndexInput;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.startsWith;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ThreadLeakFilters(filters = CleanerDaemonThreadLeakFilter.class)
+public class CompositeDirectoryTests extends BaseRemoteSegmentStoreDirectoryTests {
+    private FileCache fileCache;
+    private FSDirectory localDirectory;
+    private CompositeDirectory compositeDirectory;
+
+    @Before
+    public void setup() throws IOException {
+        setupRemoteSegmentStoreDirectory();
+        localDirectory = mock(FSDirectory.class);
+        fileCache = mock(FileCache.class);
+        compositeDirectory = new CompositeDirectory(localDirectory, remoteSegmentStoreDirectory, fileCache);
+    }
+
+    public void testListAll() throws IOException {
+        when(localDirectory.listAll()).thenReturn(new String[]{});
+        String[] actualFileNames = compositeDirectory.listAll();
+        String[] expectedFileNames = new String[] {};
+        assertArrayEquals(expectedFileNames, actualFileNames);
+
+        populateMetadata();
+        when(localDirectory.listAll()).thenReturn(new String[] { "_1.cfe", "_2.cfe", "_0.cfe_block_7", "_0.cfs_block_7" });
+
+        actualFileNames = compositeDirectory.listAll();
+        expectedFileNames = new String[] { "_0.cfe", "_0.cfs", "_0.si", "_1.cfe", "_2.cfe", "segments_1" };
```

Comment:
why are the block files no included here?  I would expect listAll should show everything in the directory even if it is not part of the sis metadata -> `SegmentInfos.files()`